### ISSUE TITLE
HookInstall access violation crash

### ISSUE THREAD

**Issue by CrosRoad95** [multitheftauto/mtasa-blue#985]
**Describe the bug**
 0xC0000005: Access violation writing location 0x00000000.

I think it is reason of crash
`HookInstall(HOOKPOS_CAEAmbienceTrackManager_CheckForPause, (DWORD)HOOK_CAEAmbienceTrackManager_CheckForPause, 6);`
in `Client\game_sa\CAudioEngineSA.cpp`

**To reproduce**
unknown

**Expected behaviour**
Crash shouldn't happened

**Screenshots**
stack frame
![image](https://user-images.githubusercontent.com/22455534/59156274-991d3100-8a98-11e9-9465-fcc7ea586061.png)

Exception got handled in
![image](https://user-images.githubusercontent.com/22455534/59156290-e1d4ea00-8a98-11e9-81bf-c9014e26b1ef.png)

**Version**
mta 1.5.6 release 18427

**Additional context**

**Comment by ghost** [comment]
GTA SA uses a secondary thread of loading game files. I think we can remove assert here?

**Comment by qaisjp** [comment]
I don't think that the error message `0xC0000005: Access violation writing location 0x00000000` is an assertion failure here.

Screenshotted line can be found [here](https://github.com/multitheftauto/mtasa-blue/blob/48878af361c97c840dd267e654f5247b5a513c18/Shared/sdk/SharedUtil.MemAccess.hpp#L88).

> GTA SA uses a secondary thread of loading game files. I think we can remove assert here?

cc @ccw808 since they wrote this code, see https://github.com/multitheftauto/mtasa-blue/commit/4e28ffc347e2495d499e0027595d1701ddacb817#diff-945a79a21c9e3491c229e147f7d26393R99

**Comment by ccw808** [comment]
The stack frame shows that it is the main thread, so `SharedUtil::GetMainThreadId` must be failing sometimes. https://github.com/multitheftauto/mtasa-blue/blob/48878af361c97c840dd267e654f5247b5a513c18/Shared/sdk/SharedUtil.Misc.hpp#L1206

**Comment by ghost** [comment]
@ccw808 Any idea why the function is failing? According to msdn:
>The CreateThread and CreateRemoteThread functions also return an identifier that uniquely identifies the thread throughout the system. A thread can use the GetCurrentThreadId function to get its own thread identifier. The identifiers are valid from the time the thread is created until the thread has been terminated. Note that no thread identifier will ever be 0

**Comment by Dutchman101** [comment]
For the record, on r16588 the offset (shifted) is 0001a32f 
Basically, on each major release it occurs 400 to 500 times

> The CreateThread and CreateRemoteThread functions also return an identifier that uniquely identifies the thread throughout the system. A thread can use the GetCurrentThreadId function to get its own thread identifier. The identifiers are valid from the time the thread is created until the thread has been terminated. Note that no thread identifier will ever be 0

Perhaps some exceptional circumstance, like hardware issues, interference (corruption) from an external source, etc?

**Comment by ccw808** [comment]
> Any idea why the function is failing

Maybe `CreateToolhelp32Snapshot` is failing, or `GetThreadTimes` is returning incorrect data
